### PR TITLE
PcapNg - Apple Process Information Block

### PR DESCRIPTION
**Pull request by guedou**
This PR adds support for process information stored in a PcapNg by the `tcpdump` command on macOS.

How to test it?
```shell
sudo tcpdump -kA -r test.pcapng
```

```python
>>> l = rdpcap("test.pcapng")
>>> l[0].process_information
{'proc': {'id': 2116, 'name': 'com.apple.WebKit'}, 'eproc': {'id': 2107, 'name': 'Safari'}}
```

### CODE REVIEW

**Comment by guedou** [comment]
I welcome comments, yet this needs to be tested against several big pcap files before being merged.

File: scapy/utils.py [hunk 1]
```diff
@@ -1899,6 +1919,31 @@ def _read_block_dsb(self, block, size):
...
+                    len(block))
+            raise EOFError
+
+        # Get Options
+        options = self._read_options(block)
+        for code, value in options.items():
+            if code == 2:
+                process_information["name"] = value.decode("ascii", "backslashreplace")
+            elif code == 4:
+                process_information["uuid"] = str(UUID(bytes=value))
```
Comment (evverx) [inline scapy/utils.py:1942]: I tested it with https://github.com/secdev/scapy/pull/4378 and the fuzz target crashed eventually with
```
ValueError: bytes is not a 16-char string
Traceback (most recent call last):
  File "/home/vagrant/scapy/pcap_fuzzer.py", line 30, in TestOneInput
    scapy.utils.rdpcap(io.BytesIO(input_bytes))
  File "/home/vagrant/scapy/scapy/utils.py", line 1261, in rdpcap
    return fdesc.read_all(count=count)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vagrant/scapy/scapy/utils.py", line 1538, in read_all
    res = self._read_all(count)
          ^^^^^^^^^^^^^^^^^^^^^
  File "/home/vagrant/scapy/scapy/utils.py", line 1448, in _read_all
    p = self.read_packet()  # type: Packet
        ^^^^^^^^^^^^^^^^^^
  File "/home/vagrant/scapy/scapy/utils.py", line 1962, in read_packet
    rp = super(PcapNgReader, self)._read_packet(size=size)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vagrant/scapy/scapy/utils.py", line 1693, in _read_packet
    res = self._read_block()
          ^^^^^^^^^^^^^^^^^^
  File "/home/vagrant/scapy/scapy/utils.py", line 1620, in _read_block
    return self.blocktypes[blocktype](block, size)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vagrant/scapy/scapy/utils.py", line 1942, in _read_block_pib
    process_information["uuid"] = str(UUID(bytes=value))
                                      ^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/uuid.py", line 187, in __init__
    raise ValueError('bytes is not a 16-char string')
ValueError: bytes is not a 16-char string

==28164== ERROR: libFuzzer: fuzz target exited
```

Comment (evverx) [reply scapy/utils.py:1942]: The hexdump of its input was
```
00000000  0a 0d 0d 0a 20 00 00 00  4d 3c 2b 1a 01 00 00 00  |.... ...M<+.....|
00000010  00 14 00 00 00 4b 53 00  01 00 08 ab 20 00 00 00  |.....KS..... ...|
00000020  01 00 00 80 20 00 00 00  4d 3c 2b 1a 01 00 00 00  |.... ...M<+.....|
00000030  01 00 00 00 00 01 00 00  01 00 08 ab 20 00 00 00  |............ ...|
00000040  01 00 00 80 20 00 00 00  4d 3c 2b 1a 01 00 00 00  |.... ...M<+.....|
00000050  04 00 00 00 01 00 00 00  1a 01 00 af 20 00 00 00  |............ ...|
00000060
```

**Comment by guedou** [comment]
@evverx That's a good catch! Thanks a lot for fuzzing it. This is indeed the piece of the code that I was not able to test.

**Comment by evverx** [comment]
Hopefully the action should help to catch things like this automatically.

It seems it still crashes. As far as I understand the condition is inverted in that `UUID()` is called when the length isn't 16.

**Review by polybassa — APPROVED** [review]

**Review by gpotter2 — APPROVED** [review]

**Pull request merged**